addon/-private/state-machine/-base.ts

```diff
@@ -56,123 +56,61 @@ export default abstract class StateMachine extends EmberObject {
   }
 
   addStep(this: StateMachine, name: string) {
-    if (this.stepsToAdd.includes(name)) {
+    if (this.stepTransitions.includes(name)) {
       return;
     }
 
-    this.stepsToAdd.push(name);
-    return bind(
+    this.stepTransitions.pushObject(name);
+
+    bind(
```

Comment:
You don't need `bind` here at all. This should just be

```javascript
scheduleOnce('actions', this, this.updateFirstCurrentAndLastSteps)
```

just like the original implementation that I had.